components/Banner/examples/BannerTypicalUseExampleViewController.m

```diff
@@ -129,230 +130,238 @@ + (NSDictionary *)catalogMetadata {
+
   return [bannerExampleList copy];
 }
 
 #pragma mark - Example handlers
 
 - (void)showSingleLineStyleBanner {
-  if (self.bannerViewContainer) {
-    [self.bannerViewContainer removeFromSuperview];
+  if (self.bannerView) {
+    [self.bannerView removeFromSuperview];
   }
-  // Set up banner view container
-  UIView *bannerViewContainer = [[UIView alloc] initWithFrame:CGRectZero];
-  [self.view addSubview:bannerViewContainer];
-  bannerViewContainer.backgroundColor = [UIColor whiteColor];
-  self.bannerViewContainer = bannerViewContainer;
 
   MDCBannerView *bannerView = [[MDCBannerView alloc] init];
-  bannerView.image = [UIImage imageNamed:@"Email"];
-  bannerView.text = exampleShortText;
-
-  MDCButton *button = [bannerView.buttons firstObject];
-  [button setTitle:@"DISMISS" forState:UIControlStateNormal];
-  UIFont *buttonFont = [MDCTypography body2Font];
-  [button setTitleFont:buttonFont forState:UIControlStateNormal];
-  [button setTitleFont:buttonFont forState:UIControlStateHighlighted];
+  bannerView.textLabel.text = exampleShortText;
+  bannerView.backgroundColor = self.colorScheme.surfaceColor;
+  UIEdgeInsets margins = UIEdgeInsetsZero;
+  margins.left = exampleBannerContentPadding;
+  margins.right = exampleBannerContentPadding;
+  bannerView.layoutMargins = margins;
+  [self.view addSubview:bannerView];
+  self.bannerView = bannerView;
+
+  MDCButton *button = bannerView.leadingButton;
+  [button setTitle:@"Dismiss" forState:UIControlStateNormal];
+  button.uppercaseTitle = YES;
   [button setTitleColor:self.colorScheme.primaryColor forState:UIControlStateNormal];
   button.backgroundColor = self.colorScheme.surfaceColor;
+  bannerView.trailingButton.hidden = YES;
 
-  [self.bannerViewContainer addSubview:bannerView];
   [button addTarget:self
                 action:@selector(dismissBanner)
       forControlEvents:UIControlEventTouchUpInside];
-  CGSize bannerViewSize =
-      [bannerView sizeThatFits:CGSizeMake(exampleBannerContentSingleLineWidth, CGFLOAT_MAX)];
-  bannerView.frame = CGRectMake(0.0f, 0.0f, bannerViewSize.width, bannerViewSize.height);
 
   // Adjust bannerViewContainer's frame
   CGFloat topAreaInset = 0.0f;
   if (@available(iOS 11.0, *)) {
     topAreaInset = self.view.safeAreaInsets.top;
   }
-  self.bannerViewContainer.frame = CGRectMake(
-      0.0f, topAreaInset, [UIScreen mainScreen].bounds.size.width, bannerViewSize.height);
-  bannerView.center = CGPointMake(self.bannerViewContainer.frame.size.width / 2,
-                                  self.bannerViewContainer.frame.size.height / 2);
+
+  CGSize bannerViewSize =
+      [bannerView sizeThatFits:CGSizeMake(CGRectGetWidth(self.view.bounds), CGFLOAT_MAX)];
+  bannerView.frame = CGRectMake(0.0f, topAreaInset, bannerViewSize.width, bannerViewSize.height);
+}
+
+- (void)showSingleLineStyleBannerWithIcon {
+  [self showSingleLineStyleBanner];
+  // TODO: https://github.com/material-components/material-components-ios/issues/6838
```

Comment:
Please add a comment describing what the TODO is. The link is helpful to track the work, but there should be an explanation of the bug/issue/clean-up as well.

Reply:
Sure. This TODO was done. Addressed in 808dd2535cf1f193b23024da6c1c20b98092bed9.